>From 0eafe5ddadee8e690672b452121eb8cf88736293 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem.jover@nokia.com>
Date: Tue, 28 Oct 2008 18:46:28 +0200
Subject: [PATCH] lvm: Fix possible NULL value handling

Add a missing NULL check, and correct them by moving the pointer
operations after the actual check.
---
 disk/lvm.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/disk/lvm.c b/disk/lvm.c
index cd9e447..df8c99b 100644
--- a/disk/lvm.c
+++ b/disk/lvm.c
@@ -373,9 +373,10 @@ grub_lvm_scan_device (const char *name)
 	      grub_memcpy (pv->name, p, s);
 	      pv->name[s] = '\0';
 	      
-	      p = grub_strstr (p, "id = \"") + sizeof("id = \"") - 1;
+	      p = grub_strstr (p, "id = \"");
 	      if (p == NULL)
 		goto pvs_fail;
+	      p += sizeof("id = \"") - 1;
 	      
 	      grub_memcpy (pv->id, p, GRUB_LVM_ID_STRLEN);
 	      pv->id[GRUB_LVM_ID_STRLEN] = '\0';
@@ -387,7 +388,10 @@ grub_lvm_scan_device (const char *name)
 	      pv->next = vg->pvs;
 	      vg->pvs = pv;
 	      
-	      p = grub_strchr (p, '}') + 1;
+	      p = grub_strchr (p, '}');
+	      if (p == NULL)
+		goto pvs_fail;
+	      p++;
 	      
 	      continue;
 	    pvs_fail:
-- 
1.6.0.2

